openmc/data/decay.py

```diff
@@ -0,0 +1,484 @@
+        string, it is assumed to be the filename for the ENDF file.
+
+    Attributes
+    ----------
+    cumulative : list of dict
+        Cumulative yields for each tabulated energy. Each item in the list is a
+        dictionary whose keys are nuclide names and values are cumulative
+        yields. The i-th dictionary corresponds to the i-th incident neutron
+        energy.
+    energies : Iterable of float or None
+        Energies at which fission product yields are tabulated.
+    independent : list of dict
+        Independent yields for each tabulated energy. Each item in the list is a
+        dictionary whose keys are nuclide names and values are independent
+        yields. The i-th dictionary corresponds to the i-th incident neutron
+        energy.
+    nuclide : dict
+        Properties of the fissioning nuclide.
+
+    Notes
+    -----
+    Neutron fission yields are typically not measured with a monoenergetic
+    source of neutrons. As such, if the fission yields are given at, e.g.,
+    0.0253 eV, one should interpret this as meaning that they are derived from a
+    typical thermal reactor flux spectrum as opposed to a monoenergetic source
+    at 0.0253 eV.
+
+    """
+    def __init__(self, ev_or_filename):
+        # Define function that can be used to read both independent and
+        # cumulative yields
+        def get_yields(file_obj):
+            # Determine number of energies
+            n_energy = get_head_record(file_obj)[2]
+            energies = np.zeros(n_energy)
+
+            data = []
+            for i in range(n_energy):
+                # Determine i-th energy and number of products
+                items, values = get_list_record(file_obj)
+                energies[i] = items[0]
+                n_products = items[5]
+
+                # Get yields for i-th energy
+                yields = {}
+                for j in range(n_products):
+                    Z, A = divmod(int(values[4*j]), 1000)
+                    isomeric_state = int(values[4*j + 1])
+                    name = ATOMIC_SYMBOL[Z] + str(A)
+                    if isomeric_state > 0:
+                        name += '_m{}'.format(isomeric_state)
+                    yield_j = ufloat(values[4*j + 2], values[4*j + 3])
+                    yields[name] = yield_j
+
+                data.append(yields)
+
+            return energies, data
+
+        # Get evaluation if str is passed
+        if isinstance(ev_or_filename, Evaluation):
+            ev = ev_or_filename
+        else:
+            ev = Evaluation(ev_or_filename)
+
+        # Assign basic nuclide properties
+        self.nuclide = {
+            'name': ev.gnd_name,
+            'atomic_number': ev.target['atomic_number'],
+            'mass_number': ev.target['mass_number'],
+            'isomeric_state': ev.target['isomeric_state']
+        }
+
+        # Read independent yields
+        if (8, 454) in ev.section:
+            file_obj = StringIO(ev.section[8, 454])
```

Comment:
This buffer is never closed.

https://docs.python.org/3/library/io.html#io.IOBase.close